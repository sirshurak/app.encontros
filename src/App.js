import React, { useState, useRef, useCallback } from "react";
import Navigator from "react.cordova-navigation_controller";
import "bootstrap/dist/css/bootstrap.min.css";
// Learn more about the Navigator: https://www.npmjs.com/package/react.cordova-navigation_controller

import "./app.css";

import AppContext from "./contexts/AppContext";

// --pages--//
import Home from "./pages/Home/index";
import UsersDetail from "./pages/Users/Detail";

function App() {
  const [currentPage, setCurrentPage] = useState("Home");
  const [currentData, setCurrentData] = useState();
  const navigator = useRef();

  const menuClick = useCallback(
    (goToPage) => {
      if (navigator.current) navigator.current.changePage(goToPage);
    },
    [navigator]
  );

  const changeCurrentPageCallback = useCallback(
    (page) => {
      setCurrentPage(page);
    },
    [setCurrentPage]
  );

  const currentActive = (page) => (currentPage === page ? "active" : undefined);
  const getCurrentData = (page) =>
    currentPage === page ? currentData : undefined;

  const changePageCallback = useCallback(
    (page, data) => {
      setCurrentData(data);
      menuClick(page);
    },
    [setCurrentData, menuClick]
  );

  const providerValue = { changePage: changePageCallback };

  return (
    <div className="App">
      <AppContext.Provider value={providerValue}>
        <main>
          <Navigator
            onRef={(ref) => {
              navigator.current = ref;
            }} // Required
            top="0px"
            onChangePage={changeCurrentPageCallback}
          >
            <Home
              key="Home"
              levelPage={0}
              data={getCurrentData("Home")}
              top="0px"
              backgroundColor="inherit"
            />
            <UsersDetail
              key="UsersDetail"
              backgroundColor="inherit"
              levelPage={1}
              data={getCurrentData("UsersDetail")}
            />
          </Navigator>
        </main>
        <footer className="footer-menu">
          <ul>
            <li>
              <button
                onClick={() => menuClick("Home")}
                className={currentActive("Home")}
                type="button"
              >
                Conversas
              </button>
            </li>
          </ul>
        </footer>
      </AppContext.Provider>
    </div>
  );
}

export default App;
