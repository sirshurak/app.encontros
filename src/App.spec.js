import { shallow, mount } from 'enzyme';
import App from './App';

jest.mock('./hooks/UseUsers');

import useUsers from './hooks/UseUsers';

describe('App component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<App />)
        expect(component).toMatchSnapshot();
    });

    it('should mount and change page', () => {
        useUsers.mockReturnValue({
            users: [{
                photo: 'https://picsum.photos/200/?image=26',
                username: 'username',
                city: 'Test City',
                gender: 'Man',
                sexualOrientation: 'Heterosexual',
                age: 27
            }]
        });
        document.getElementById = jest.fn(() => ({ addEventListener: jest.fn((name, func) => func()), removeEventListener: jest.fn() }));
        const component = mount(<App />);
        
        component.find('button').forEach((button) => button.simulate('click', {}));
    });
});