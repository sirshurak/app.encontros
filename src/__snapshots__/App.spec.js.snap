// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match shallow snapshop 1`] = `
<div
  className="App"
>
  <ContextProvider
    value={
      Object {
        "changePage": [Function],
      }
    }
  >
    <main>
      <Navigator
        onChangePage={[Function]}
        onRef={[Function]}
        top="0px"
      >
        <Home
          backgroundColor="inherit"
          key="Home"
          levelPage={0}
          top="0px"
        />
        <UsersDetail
          backgroundColor="inherit"
          key="UsersDetail"
          levelPage={1}
        />
      </Navigator>
    </main>
    <footer
      className="footer-menu"
    >
      <ul>
        <li>
          <button
            className="active"
            onClick={[Function]}
            type="button"
          >
            Conversas
          </button>
        </li>
      </ul>
    </footer>
  </ContextProvider>
</div>
`;
